service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -115,13 +118,103 @@ public class EhrExtractStatusService {
-    private static final String REASON_ERROR_CODE = "04";
+    private static final String REASON_ERROR_CODE = "99";
     public static final String REASON_ERROR_MESSAGE = "The acknowledgement has been received after 8 days";
 
     private final MongoTemplate mongoTemplate;
     private final EhrExtractStatusRepository ehrExtractStatusRepository;
     private final TimestampService timestampService;
 
+    @Scheduled(cron = "${timeout.cronTime}")
+    public void checkForEhrExtractAckTimeouts() {
+        List<EhrExtractStatus> inProgressEhrExtractTransfers = findInProgressTransfers();
+        var now = Instant.now();
+        var ehrExtractStatusWithExceededUpdateLimit = inProgressEhrExtractTransfers
+            .stream()
+            .filter(ehrExtractStatus -> Objects.isNull(ehrExtractStatus.getEhrReceivedAcknowledgement()))
+            .filter(ehrExtractStatus -> hasLastUpdateExceededEightDays(ehrExtractStatus, now))
+            .toList();
+
+        if (!ehrExtractStatusWithExceededUpdateLimit.isEmpty()) {
+            updateEhrExtractStatusListWithEhrReceivedAcknowledgementError(ehrExtractStatusWithExceededUpdateLimit,
+                                                                          REASON_ERROR_CODE,
+                                                                          REASON_ERROR_MESSAGE);
+        }
+    }
+
+    boolean hasEhrStatusReceivedAckWithErrors(String conversationId) {
+
+        var ehrExtractStatus = fetchEhrExtractStatus(conversationId, "NACK");
+
+        var ehrReceivedAcknowledgement = ehrExtractStatus.getEhrReceivedAcknowledgement();
+        if (ehrReceivedAcknowledgement == null) {
+            return false;
+        }
+
+        var errors = ehrReceivedAcknowledgement.getErrors();
+        if (errors == null || errors.isEmpty()) {
+            return false;
+        }
+
+        var ehrReceivedAckErrorDetailsThatContainsSearchedErrCodeAndMsg = errors.stream()
+            .filter(err -> REASON_ERROR_CODE.equals(err.getCode()) && REASON_ERROR_MESSAGE.equals(err.getDisplay()))
+            .findAny();
+
+        return ehrReceivedAckErrorDetailsThatContainsSearchedErrCodeAndMsg.isPresent();
+    }
+
+    void updateEhrExtractStatusListWithEhrReceivedAcknowledgementError(List<EhrExtractStatus> ehrExtractStatusList,
+                                                                       String errorCode,
+                                                                       String errorMessage) {
+
+        ehrExtractStatusList.stream().forEach(ehrExtractStatus -> {
+            try {
+                updateEhrExtractStatusWithEhrReceivedAckError(ehrExtractStatus.getConversationId(), errorCode, errorMessage);
+            } catch (Exception e) {
+
+                logger().error("An error occurred when closing a failed process for conversation_id: {}",
+                               ehrExtractStatus.getConversationId(), e);
+                throw e;
+            }
```

Comment:
Are there any more specific exceptions we can catch here rather than using `Exception.java`? Furthermore could we please rename `e` to `exception` 😄

Small, but it all adds up imo!